spec/models/user_spec.rb

```diff
@@ -6,92 +6,149 @@
+
+      it 'has many inverse_friends' do
+        assc = described_class.reflect_on_association(:inverse_friends)
+        expect(assc.macro).to eq :has_many
+      end
+
+    end
+
+      
+    context 'Posts' do
+      it 'has many posts' do
+        assc = described_class.reflect_on_association(:posts)
+        expect(assc.macro).to eq :has_many
+      end
+      
+      it 'returns last post when called' do
+        user.posts << post
+        expect(user.posts.last).to eq(post)
+      end
+      
+      it 'is expected to destroy dependent posts' do
+        post
+        expect { post.user.destroy }.to change { Post.count }.by(-1)
+      end
+    end
+
+    context 'Comments' do
+      it 'has many comments' do
+        assc = described_class.reflect_on_association(:comments)
+        expect(assc.macro).to eq :has_many
+      end
+      
+      it 'returns comments when called upon comments method' do
+        user.comments << comment
+        expect(user.comments.last).to eq(comment)
+      end
+
+      it 'is depedent on user deletion' do
+        user.comments << comment
+        user.destroy
+        expect(Comment.count).to eq 0
+      end
+    end
+
+    context 'Likes' do
+      it 'has many likes' do
+        assc = described_class.reflect_on_association(:likes)
+        expect(assc.macro).to eq :has_many
+      end
+      
+      it 'increases number of comments likes by one when user likes'  do
+        user.liked_comments << comment
+        expect(user.liked_comments.count).to eq 1
+      end
+
+      it 'increases number of posts likes by one when user likes'  do
+        user.liked_posts << post
+        expect(user.liked_posts.count).to eq 1
+      end
+
+      it 'is expected to destroy dependent likes' do
+        user.liked(post)
+        expect { user.destroy }.to change { Like.count }.by(-1)
+      end
     end
     
-    it 'returns all posts when called' do
-      user.posts << post
-      expect(user.posts.last).to eq(post)
-    end
-
-    it 'is expected to destroy dependent posts' do
-      post
-      expect { post.user.destroy }.to change { Post.count }.by(-1)
-    end
-
-    it 'has many comments' do
-      assc = described_class.reflect_on_association(:comments)
-      expect(assc.macro).to eq :has_many
-    end
-
-    it 'returns comments when called upon comments method' do
-      user.comments << comment
-      expect(user.comments.last).to eq(comment)
-    end
-
-    it 'has many likes' do
-      assc = described_class.reflect_on_association(:likes)
-      expect(assc.macro).to eq :has_many
-    end
-
-    it 'returns likes when called upon likes method' do
-      user.likes << like
-      expect(user.likes.last).to eq(like)
-    end
-
   end
 
-  context 'validations' do
-    it 'is invalid if username is missing' do
-      user.username = nil
-      expect(user.valid?).to be false
+  describe 'Validations' do
+    context 'missing username' do
+      it 'is invalid' do
+        user.username = nil
```

Comment:
I observe the ff line:
```ruby
 let(:user) { create :user }
```
To use this variable for a validation spec we should not be using `create`, but `build`. In the actual models, validation occurs prior to the record being saved. So let's simulate that here.